nmap.go

```diff
@@ -394,6 +316,100 @@ func (s *Scanner) RunAsync() error {
 	return nil
 }
 
+// RunDirect runs nmap synchronously. The XML output is written directly to a file and stdout is written to the console.
+func (s *Scanner) RunDirect(file string) (result *Run, warnings []string, err error) {
+	var stderr bytes.Buffer
+
+	args := s.args
+
+	// Enable XML output
+	args = append(args, "-oX")
+
+	// Get XML output in stdout instead of writing it in a file
+	args = append(args, file)
+
+	// Prepare nmap process
+	cmd := exec.Command(s.binaryPath, args...)
+	if s.modifySysProcAttr != nil {
+		s.modifySysProcAttr(cmd.SysProcAttr)
+	}
+	cmd.Stderr = &stderr
+	cmd.Stdout = os.Stdout
+
+	// Run nmap process
+	err = cmd.Start()
+	if err != nil {
+		return nil, warnings, err
+	}
+
+	// Make a goroutine to notify the select when the scan is done.
+	done := make(chan error, 1)
+	go func() {
+		done <- cmd.Wait()
+	}()
+
+	// Wait for nmap process or timeout
+	select {
+	case <-s.ctx.Done():
+
+		// Context was done before the scan was finished.
+		// The process is killed and a timeout error is returned.
+		_ = cmd.Process.Kill()
+
+		return nil, warnings, ErrScanTimeout
+	case <-done:
+		xml, err := os.ReadFile(file)
+		if err != nil {
+			return nil, warnings, err
+		}
+
+		return s.processOutput(xml, stderr.Bytes())
+	}
+}
+
+func (s *Scanner) processOutput(xml []byte, stderr []byte) (result *Run, warnings []string, err error) {
+	// Process nmap stderr output containing none-critical errors and warnings
+	// Everyone needs to check whether one or some of these warnings is a hard issue in their use case
```

Comment:
Please end this sentences with a dot
```suggestion
	// Process nmap stderr output containing none-critical errors and warnings.
	// Everyone needs to check whether one or some of these warnings is a hard issue in their use case.
```